### PR TITLE
Fix cmd/broker unit test flakiness 

### PR DESCRIPTION
**Pull request by piotrmiskiewicz**
**Description**

Changes proposed in this pull request:
- sometimes update tests were failing because after the update (PATCH) the reconciler fake client was guided to a response for the previous operation (which has finished) and the fresh reconciler operation was never ended.

The change was to program the fake client with a response before any interaction. 

**Related issue(s)**

### CODE REVIEW

File: internal/reconciler/fake.go [hunk 1]
```diff
@@ -61,15 +70,40 @@ func (c *FakeClient) GetCluster(clusterName string, configVersion int64) (*recon
...
+		msg := intKeys(existingCluster.clusterStates)
+		return &reconcilerApi.HTTPClusterResponse{}, fmt.Errorf("cluster state not found, version=%d (%s)", configVersion, msg)
+	}
+	if existingCluster.numberOfGetStatusCalls == 0 && c.expectedStatus != nil {
+		state.Status = *c.expectedStatus
 	}
 	return state, nil
 }
 
+func intKeys(states map[int64]*reconcilerApi.HTTPClusterResponse) string {
```
Comment (jaroslaw-pieszka) [inline internal/reconciler/fake.go:91]: Do we just join all states from states map?

File: internal/reconciler/fake.go [hunk 2]
```diff
@@ -61,15 +70,40 @@ func (c *FakeClient) GetCluster(clusterName string, configVersion int64) (*recon
...
+	result := ""
+	for k, _ := range states {
+		result = result + fmt.Sprintf("[%d] ", k)
+	}
+	return result
+}
+
+func stringKeys(clusters map[string]*registeredCluster) string {
```
Comment (jaroslaw-pieszka) [inline internal/reconciler/fake.go:99]: And do we here join all clusters from inventoryClusters?

File: internal/reconciler/fake.go (hunk 1 above)
Comment (piotrmiskiewicz) [reply internal/reconciler/fake.go:91]: I've added states

**Review by jaroslaw-pieszka — APPROVED** [review]

**Pull request merged**